providers/from-env/pkg/provider.go

```diff
@@ -0,0 +1,141 @@
+package from_env
+
+import (
+	"github.com/open-feature/golang-sdk/pkg/openfeature"
+)
+
+// FromEnvProvider implements the FeatureProvider interface and provides functions for evaluating flags
+type FromEnvProvider struct {
+	envFetch envFetch
+}
+
+const (
+	ReasonError          = "ERROR"
+	ReasonStatic         = "STATIC"
+	ReasonTargetingMatch = "TARGETING_MATCH"
+
+	ErrorTypeMismatch = "TYPE_MISMATCH"
+	ErrorParse        = "PARSE_ERROR"
+	ErrorFlagNotFound = "FLAG_NOT_FOUND"
+)
+
+// Metadata returns the metadata of the provider
+func (p *FromEnvProvider) Metadata() openfeature.Metadata {
+	return openfeature.Metadata{
+		Name: "from-env-flag-evaluator",
+	}
+}
+
+// Hooks returns hooks
+func (p *FromEnvProvider) Hooks() []openfeature.Hook {
+	return []openfeature.Hook{}
+}
+
+// BooleanEvaluation returns a boolean flag
+func (p *FromEnvProvider) BooleanEvaluation(flagKey string, defaultValue bool, evalCtx openfeature.EvaluationContext, _ openfeature.EvaluationOptions) openfeature.BoolResolutionDetail {
```

Comment:
We should probably hold off on merging these and update to remove EvaluationOptions once https://github.com/open-feature/golang-sdk/pull/48 has been merged

Reply:
these changes have been merged, examples and implementation have been updated :)